From 4fc564072d4e27d91e090b98ce28e23994d86766 Mon Sep 17 00:00:00 2001
From: Ganesh Biradar <gbiradar@intrinsyc.com>
Date: Wed, 30 Jan 2019 11:28:27 +0530
Subject: [PATCH 325/334] [8278] Added ABS_X and ABS_Y params

Userpace Xserver evdev driver will recognise touchscreen with ABS_X and ABS_Y parameter

Change-Id: Icd2da155ef74117c9173d90fa2491812146b82e5
Signed-off-by: Ganesh Biradar <gbiradar@intrinsyc.com>
(cherry picked from commit 13022689e063ee2722536d53117c78fb1e3000b1)
---
 drivers/input/touchscreen/gt9xx/gt9xx.c | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/drivers/input/touchscreen/gt9xx/gt9xx.c b/drivers/input/touchscreen/gt9xx/gt9xx.c
index 04bde898ec1e..33ef1ea53509 100644
--- a/drivers/input/touchscreen/gt9xx/gt9xx.c
+++ b/drivers/input/touchscreen/gt9xx/gt9xx.c
@@ -468,6 +468,8 @@ static void gtp_type_a_report(struct goodix_ts_data *ts, u8 touch_num,
 				input_report_key(ts->input_dev,
 							BTN_TOOL_FINGER, true);
 			}
+			input_report_abs(ts->input_dev, ABS_X, points->x);
+			input_report_abs(ts->input_dev, ABS_Y, points->y);
 			input_report_abs(ts->input_dev, ABS_MT_POSITION_X,
 					 points->x);
 			input_report_abs(ts->input_dev, ABS_MT_POSITION_Y,
@@ -521,6 +523,8 @@ static void gtp_mt_slot_report(struct goodix_ts_data *ts, u8 touch_num,
 				input_mt_report_slot_state(ts->input_dev,
 						MT_TOOL_FINGER, true);
 			}
+			input_report_abs(ts->input_dev, ABS_X, points->x);
+			input_report_abs(ts->input_dev, ABS_Y, points->y);
 			input_report_abs(ts->input_dev, ABS_MT_POSITION_X,
 					 points->x);
 			input_report_abs(ts->input_dev, ABS_MT_POSITION_Y,
@@ -1632,8 +1636,8 @@ static s8 gtp_request_input_dev(struct goodix_ts_data *ts)
 		dev_info(&ts->client->dev, "Use type A report protocol\n");
 	}
 
-	input_set_capability(ts->input_dev, EV_KEY, GTP_PEN_BUTTON1);
-	input_set_capability(ts->input_dev, EV_KEY, GTP_PEN_BUTTON2);
+/*	input_set_capability(ts->input_dev, EV_KEY, GTP_PEN_BUTTON1);
+	input_set_capability(ts->input_dev, EV_KEY, GTP_PEN_BUTTON2);*/
 
 	/* touch key register */
 	for (index = 0; index < ts->pdata->key_nums; index++)
@@ -1646,6 +1650,10 @@ static s8 gtp_request_input_dev(struct goodix_ts_data *ts)
 	if (ts->pdata->swap_x2y)
 		GTP_SWAP(ts->pdata->abs_size_x, ts->pdata->abs_size_y);
 
+	input_set_abs_params(ts->input_dev, ABS_X, 0,
+			     ts->pdata->abs_size_x, 0, 0);
+	input_set_abs_params(ts->input_dev, ABS_Y, 0,
+			     ts->pdata->abs_size_y, 0, 0);
 	input_set_abs_params(ts->input_dev, ABS_MT_POSITION_X, 0,
 			     ts->pdata->abs_size_x, 0, 0);
 	input_set_abs_params(ts->input_dev, ABS_MT_POSITION_Y, 0,
-- 
2.24.1

